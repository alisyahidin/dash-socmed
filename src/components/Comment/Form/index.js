import React from 'react'
import Avatar from 'react-avatar'
import { Button, Form, Col } from 'react-bootstrap'

const CommentForm = () => (
  <Form.Row className="w-100 m-0">
    <Col md={1} className="p-0 d-flex align-items-center justify-content-center">
      <Avatar size="35px" name="Ali Syahidin" round />
    </Col>
    <Col md={9}>
      <Form.Control placeholder="Write comment . . ." />
    </Col>
    <Col className="d-flex align-items-center">
      <Button className="w-100" variant="outline-primary" type="submit">
        Send
      </Button>
    </Col>
  </Form.Row>
)

export default CommentForm