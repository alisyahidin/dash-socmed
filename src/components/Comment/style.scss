.comment {
  &__author {
    font-size: 0.8rem;
    color: #888;
    &:hover {
      text-decoration: none;
      color: #666;
    }
  }
  &__body {
  }
}