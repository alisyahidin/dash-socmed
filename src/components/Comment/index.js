import React from 'react'
import { Link } from 'react-router-dom'
import Avatar from 'react-avatar'
import { Row, Col } from 'react-bootstrap'
import './style.scss'

const PostDetail = props => (
  <Row className="w-100 m-0 mb-2">
    <Col md={1} className="p-0 d-flex align-items-center justify-content-center">
      <Avatar size="35px" name="Ali Syahidin" round />
    </Col>
    <Col className="p-0 pl-2 d-flex flex-column">
      <Link to="/users/3" className="comment__author">
        @username
      </Link>
      <p className="comment__body">First Comment !!</p>
    </Col>
  </Row>
)

export default PostDetail