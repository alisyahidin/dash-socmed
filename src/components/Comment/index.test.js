import React from 'react'
import { shallow, mount } from 'enzyme'
import Comment from './index'
import CommentForm from './Form/'

it('render <Comment /> correctly', () => {
  const component = shallow(<Comment />)

  expect(component).toMatchSnapshot()
})

it('render <CommentForm /> correctly', () => {
  const component = shallow(<CommentForm />)

  expect(component).toMatchSnapshot()
})