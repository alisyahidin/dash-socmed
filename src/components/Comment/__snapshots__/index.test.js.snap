// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Comment /> correctly 1`] = `
<ForwardRef(Bootstrap(Row))
  className="w-100 m-0 mb-2"
>
  <ForwardRef(Bootstrap(Col))
    className="p-0 d-flex align-items-center justify-content-center"
    md={1}
  >
    <withAvatarConfig
      name="Ali Syahidin"
      round={true}
      size="35px"
    />
  </ForwardRef(Bootstrap(Col))>
  <ForwardRef(Bootstrap(Col))
    className="p-0 pl-2 d-flex flex-column"
  >
    <Link
      className="comment__author"
      replace={false}
      to="/users/3"
    >
      @username
    </Link>
    <p
      className="comment__body"
    >
      First Comment !!
    </p>
  </ForwardRef(Bootstrap(Col))>
</ForwardRef(Bootstrap(Row))>
`;

exports[`render <CommentForm /> correctly 1`] = `
<ForwardRef(Bootstrap(FormRow))
  className="w-100 m-0"
>
  <ForwardRef(Bootstrap(Col))
    className="p-0 d-flex align-items-center justify-content-center"
    md={1}
  >
    <withAvatarConfig
      name="Ali Syahidin"
      round={true}
      size="35px"
    />
  </ForwardRef(Bootstrap(Col))>
  <ForwardRef(Bootstrap(Col))
    md={9}
  >
    <ForwardRef(ContextTransform)
      placeholder="Write comment . . ."
    />
  </ForwardRef(Bootstrap(Col))>
  <ForwardRef(Bootstrap(Col))
    className="d-flex align-items-center"
  >
    <ForwardRef(Bootstrap(Button))
      className="w-100"
      type="submit"
      variant="outline-primary"
    >
      Send
    </ForwardRef(Bootstrap(Button))>
  </ForwardRef(Bootstrap(Col))>
</ForwardRef(Bootstrap(FormRow))>
`;
